logging/jsonlog/jsonlog_test.go

```diff
@@ -0,0 +1,369 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jsonlog
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/logging"
+	"github.com/google/go-cmp/cmp"
+	ltype "google.golang.org/genproto/googleapis/logging/type"
+)
+
+func TestLoggerLevel(t *testing.T) {
+	buf := bytes.NewBuffer(nil)
+	l, err := NewLogger("projects/test")
+	if err != nil {
+		t.Fatalf("unable to create logger: %v", err)
+	}
+	l.w = buf
+	now := time.Date(1977, time.May, 25, 0, 0, 0, 0, time.UTC)
+	fnow := now.Format(time.RFC3339)
+	l.now = func() time.Time {
+		return now
+	}
+	e := &entry{
+		Message:   "Hello, World!",
+		Timestamp: fnow,
+	}
+
+	e.Severity = "DEBUG"
+	b, err := json.Marshal(e)
+	if err != nil {
+		t.Fatalf("unable to marshal: %v", err)
+	}
+	l.Debugf("Hello, %s!", "World")
+	if diff := cmp.Diff(string(b), strings.TrimSpace(buf.String())); diff != "" {
+		t.Fatalf("Logger.Debugf() mismatch (-want +got):\n%s", diff)
+	}
+	buf.Reset()
+
+	e.Severity = "INFO"
+	b, err = json.Marshal(e)
+	if err != nil {
+		t.Fatalf("unable to marshal: %v", err)
+	}
+	l.Infof("Hello, %s!", "World")
+	if diff := cmp.Diff(string(b), strings.TrimSpace(buf.String())); diff != "" {
+		t.Fatalf("Logger.Infof() mismatch (-want +got):\n%s", diff)
+	}
+	buf.Reset()
+
+	e.Severity = "NOTICE"
```

Comment:
Since these unit tests are repetitive, maybe we can save some lines and put test cases in an struct array? e.g. https://github.com/googleapis/google-cloud-go/blob/812ef1ffdce2e87570660b58f0e725ad51f68546/logging/logging_unexported_test.go#L298

Reply:
oh i see you did this already in TestLoggerWithLabels